From 851d420bdfd57ff43aa831c5231f434e72f7b352 Mon Sep 17 00:00:00 2001
From: Quentin Schulz <quentin.schulz@bootlin.com>
Date: Wed, 25 Jul 2018 22:54:35 +0200
Subject: [PATCH 4/4] fix bitbake warnings

WARNING: Variable key FILES_${KERNEL_PACKAGE_NAME}-devicetree (/${KERNEL_IMAGEDEST}/*.dtb /${KERNEL_IMAGEDEST}/*.dtbo) replaces original key FILES_kernel-devicetree (/${KERNEL_IMAGEDEST}/*.itb).
WARNING: Variable key RDEPENDS_${KERNEL_PACKAGE_NAME}-base (${KERNEL_PACKAGE_NAME}-image) replaces original key RDEPENDS_kernel-base ( kernel-devicetree amx3-cm3).

WARNING: Layer meta-ti should set LAYERSERIES_COMPAT_meta-ti in its conf/layer.conf file to list the core layer names it is compatible with.

Signed-off-by: Quentin Schulz <quentin.schulz@bootlin.com>
---
 conf/layer.conf                               |  2 ++
 recipes-kernel/linux/linux-ti-staging_4.14.bb | 36 +++++++++++++--------------
 2 files changed, 20 insertions(+), 18 deletions(-)

diff --git a/conf/layer.conf b/conf/layer.conf
index 8a3af7ab0bbe..170b94b33df9 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -12,6 +12,8 @@ LICENSE_PATH += "${LAYERDIR}/licenses"
 
 LAYERDEPENDS_meta-ti = "core"
 
+LAYERSERIES_COMPAT_meta-ti = "sumo"
+
 SIGGEN_EXCLUDERECIPES_ABISAFE += " \
     ti-sgx-ddk-km \
     ti-sgx-ddk-um \
diff --git a/recipes-kernel/linux/linux-ti-staging_4.14.bb b/recipes-kernel/linux/linux-ti-staging_4.14.bb
index a8d7e2805da1..6335dd4f4ac4 100644
--- a/recipes-kernel/linux/linux-ti-staging_4.14.bb
+++ b/recipes-kernel/linux/linux-ti-staging_4.14.bb
@@ -13,41 +13,41 @@ require recipes-kernel/linux/ti-uio.inc
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}-4.14:"
 
 # Pull in the devicetree files into the rootfs
-RDEPENDS_kernel-base += "kernel-devicetree"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base += "kernel-devicetree"
 
 # Add run-time dependency for PM firmware to the rootfs
-RDEPENDS_kernel-base_append_ti33x = " amx3-cm3"
-RDEPENDS_kernel-base_append_ti43x = " amx3-cm3"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_ti33x = " amx3-cm3"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_ti43x = " amx3-cm3"
 
 # Add run-time dependency for VPE VPDMA firmware to the rootfs
-RDEPENDS_kernel-base_append_dra7xx = " vpdma-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_dra7xx = " vpdma-fw"
 
 # Add run-time dependency for Goodix firmware to the rootfs
-RDEPENDS_kernel-base_append_dra7xx = " goodix-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_dra7xx = " goodix-fw"
 
 # Install boot-monitor skern file into /boot dir of rootfs
-RDEPENDS_kernel-base_append_keystone = " boot-monitor"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_keystone = " boot-monitor"
 
 # Install ti-sci-fw into /boot dir of rootfs
-RDEPENDS_kernel-base_append_k2g = " ti-sci-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_k2g = " ti-sci-fw"
 
 # Add run-time dependency for SerDes firmware to the rootfs
-RDEPENDS_kernel-base_append_keystone = " serdes-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_keystone = " serdes-fw"
 
 # Add run-time dependency for QMSS PDSP firmware to the rootfs
-RDEPENDS_kernel-base_append_keystone = " qmss-pdsp-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_keystone = " qmss-pdsp-fw"
 
 # Add run-time dependency for NETCP PA firmware to the rootfs
-RDEPENDS_kernel-base_append_k2hk = " netcp-pa-fw"
-RDEPENDS_kernel-base_append_k2e = " netcp-pa-fw"
-RDEPENDS_kernel-base_append_k2l = " netcp-pa-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_k2hk = " netcp-pa-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_k2e = " netcp-pa-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_k2l = " netcp-pa-fw"
 
 # Add run-time dependency for PRU Ethernet firmware to the rootfs
-RDEPENDS_kernel-base_append_am57xx-evm = " prueth-fw"
-RDEPENDS_kernel-base_append_am437x-evm = " prueth-fw"
-RDEPENDS_kernel-base_append_am335x-evm = " prueth-fw"
-RDEPENDS_kernel-base_append_k2g = " prueth-fw"
-RDEPENDS_kernel-base_append_am65xx-evm = " prueth-fw-bin"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_am57xx-evm = " prueth-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_am437x-evm = " prueth-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_am335x-evm = " prueth-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_k2g = " prueth-fw"
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base_append_am65xx-evm = " prueth-fw-bin"
 
 KERNEL_EXTRA_ARGS += "LOADADDR=${UBOOT_ENTRYPOINT}"
 
@@ -67,4 +67,4 @@ KERNEL_GIT_PROTOCOL = "git"
 SRC_URI += "${KERNEL_GIT_URI};protocol=${KERNEL_GIT_PROTOCOL};branch=${BRANCH} \
             file://defconfig"
 
-FILES_kernel-devicetree += "/${KERNEL_IMAGEDEST}/*.itb"
+FILES_${KERNEL_PACKAGE_NAME}-devicetree += "/${KERNEL_IMAGEDEST}/*.itb"
-- 
2.7.4

